java/maven/src/org/netbeans/modules/maven/classpath/ClassPathProviderImpl.java

```diff
@@ -672,12 +603,14 @@ public ClassPath getActiveClassPath() {
             ClassPath ret = active;
             if (ret == null) {
                 MavenProject mp = proj.getOriginalMavenProject();
-                boolean hasMainModuleDescriptor = mp.getCompileSourceRoots().stream().anyMatch((sourceRoot) -> (new File(sourceRoot, MODULE_INFO_JAVA).exists()));
+                boolean hasMainModuleDescriptor = mp.getCompileSourceRoots().stream()
+                        .anyMatch((sourceRoot) -> Files.exists(Paths.get(sourceRoot, MODULE_INFO_JAVA)));
                 if(hasMainModuleDescriptor) {
                     LOGGER.log(Level.FINER, "TestPathSelector {0} for project {1}: has main module-info.java", new Object [] {logDesc, proj.getProjectDirectory().getPath()}); // NOI18N
```

Comment:
if this code is being tuched anyway, why not removing this ugly "new Object [] {" and let the compiler do it for us?

Reply:
the advantage of this is to delay when the toString() method is called to the point the logging lib is building the message, it doesn't make a difference in this particular case but the whole class is using this way of logging. I don't really feel the need to rewrite it all.